specs/core/1_shard-data-chains.md

```diff
@@ -95,517 +94,346 @@ We define the following Python custom types for type hinting and readability:
-* Let `beacon_state` be the state where `beacon_ref.state_root == hash_tree_root(beacon_state)`. Let `prev_state` be the post-state of the `parent` if the `parent` exists, otherwise let it be `get_default_shard_state(beacon_state, shard)` (defined below). `block.core.state_root` must equal the `hash_tree_root` of the state after applying `shard_state_transition(prev_state, beacon_state, block)`.
-
-Note that these acceptance conditions depend on the canonical beacon chain; when the canonical beacon chain reorganizes, the eligibility of shard blocks should be re-evaluated.
-
-### Shard state transition function helpers
-
-```python
-def add_reward(state: ShardState, beacon_state: BeaconState, index: ValidatorIndex, delta: int) -> None:
-    epoch = compute_epoch_of_shard_slot(state.slot)
-    earlier_committee = get_period_committee(
-        beacon_state,
-        get_shard_period_start_epoch(epoch, lookback=2),
-        state.shard,
-    )
-    later_committee = get_period_committee(beacon_state, get_shard_period_start_epoch(epoch, lookback=1), state.shard)
-    if index in earlier_committee:
-        state.earlier_committee_rewards[earlier_committee.index(index)] += delta
-    elif index in later_committee:
-        state.later_committee_rewards[later_committee.index(index)] += delta
-    else:
-        raise Exception("Should never be here")
-```
+### `get_genesis_shard_block`
 
 ```python
-def add_fee(state: ShardState, beacon_state: BeaconState, index: ValidatorIndex, delta: int) -> None:
-    epoch = compute_epoch_of_shard_slot(state.slot)
-    earlier_committee = get_period_committee(beacon_state, get_shard_period_start_epoch(epoch, lookback=2), state.shard)
-    later_committee = get_period_committee(beacon_state, get_shard_period_start_epoch(epoch, lookback=1), state.shard)
-    if index in earlier_committee:
-        state.earlier_committee_fees[earlier_committee.index(index)] += delta
-    elif index in later_committee:
-        state.later_committee_fees[later_committee.index(index)] += delta
-    else:
-        raise Exception("Should never be here")
+def get_genesis_shard_block(state: BeaconState, shard: Shard) -> ShardBlock:
+    return ShardBlock(data=ShardBlockData(
+        shard=shard,
+        slot=ShardSlot(SHARD_GENESIS_EPOCH * SHARD_SLOTS_PER_EPOCH),
+        state_root=hash_tree_root(get_genesis_shard_state(state, shard)),
+    ))
 ```
 
-### Shard state transition function
+## Shard state transition function
 
 ```python
-def shard_state_transition(state: ShardState,
-                           beacon_state: BeaconState,
+def shard_state_transition(state: BeaconState,
```

Comment:
IMO it's more clear to use the explicit name `beacon_state` in phase 1 specs, just like why we use `beacon_block` in the spec. :)

Reply:
Ahh I see it now. Tricky...

Reply:
I'm a bit torn because if we stick with `state: BeaconState` then it's consistent with phase 0.

Reply:
Ok, I bit the bullet! Everything is now `beacon_state` or `shard_state` :)